blockchain/utxocache_test.go

```diff
@@ -960,3 +960,20 @@ func TestInitConsistentState(t *testing.T) {
 			blocks[len(blocks)-1].Height())
 	}
 }
+
+func TestTotalMemoryUsage(t *testing.T) {
+	// Mocked utxoCache instance
+	s := newUtxoCache(nil, 1*1024*1024)
+
+	// Call the totalMemoryUsage method
+	result := s.totalMemoryUsage()
+
+	expectedSize := uint64(1 * 1024 * 1024)
+
+	resultShoulBeTrue := result == expectedSize
+
+	// Compare the expected result with the actual result
+	if resultShoulBeTrue {
```

Comment:
It appears that there is a logical error in the condition `if resultShoulBeTrue`. This condition implies that the test should fail when the memory usage matches the expected size, which is the opposite of the intended behavior.

I recommend changing the condition to `if !resultShoulBeTrue` or ` if result != expectedSize`, indicating that the test should fail when the memory usage does not match the expected size

Reply:
Additionally, it's important to note that [totalMemoryUsage](https://github.com/btcsuite/btcd/blob/master/blockchain/utxocache.go#L253) and [maxTotalMemoryUsage](https://github.com/btcsuite/btcd/blob/master/blockchain/utxocache.go#L215) represent different concepts. [maxTotalMemoryUsage](https://github.com/btcsuite/btcd/blob/master/blockchain/utxocache.go#L215) is the maximum memory usage allowed for the cache in normal circumstances, while [totalMemoryUsage](https://github.com/btcsuite/btcd/blob/master/blockchain/utxocache.go#L253) is the map size + the size that the utxo entries are taking up.